geoh5py/shared/merging/base.py

```diff
@@ -0,0 +1,276 @@
+                        data.values,
+                        (data.entity_type, data.name, id_),
+                        data.association,
+                        previous,
+                        previous + cast(int, input_entity.n_vertices),
+                    ]
+                )
+                entity_unique_entity_types.append((data.entity_type, data.name))
+                unique_entity_types.append((data.entity_type, data.name, id_))
+                id_ += 1
+
+            # define how to define the entity_type
+            nb_keys = cls.compute_nb_keys(entity_unique_entity_types, nb_keys)
+            previous += cast(int, input_entity.n_vertices)
+
+        # create an intermediate array
+        all_data = np.array(all_data, dtype=object)
+
+        # prepare the attributes
+        unique_entity_types = list(
+            dict.fromkeys(
+                [entity_type[:nb_keys] for entity_type in unique_entity_types]  # type: ignore
+            )
+        )
+
+        return all_data, unique_entity_types, previous, nb_keys
+
+    @classmethod
+    def merge_data(cls, input_entities: list[ObjectBase]) -> dict[str, dict]:
+        """
+        Merge the data respecting the entity type, the values, and the association.
+        :param input_entities: the list of objects to merge the data from.
+        :return: a dictionary of data to add to the merged object.
+        """
+
+        # extract the data from the input entities
+        (
+            all_data,
+            unique_entity_types,
+            total_length,
+            nb_keys,
+        ) = cls.extract_data_information(input_entities)
+
+        # prepare the attributes
+        data_array, unique_associations = cls.prepare_attributes(
+            unique_entity_types, all_data, total_length, nb_keys
+        )
+
+        # create the output data
+        return cls.create_add_data_dictionary(
+            data_array, unique_entity_types, unique_associations, nb_keys
+        )
+
+    @classmethod
+    def merge_objects(
+        cls,
+        workspace: Workspace,
+        input_entities: list[ObjectBase],
+        add_data: bool = True,
+        **kwargs,
+    ):
+        """
+        Merge a list of :obj:geoh5py.objects.ObjectBase of the same type.
+        :param workspace: the workspace to use.
+        :param input_entities: the list of objects to merge.
+        :param add_data: if True, the data will be merged.
+        :param kwargs: the kwargs to create the new object.
+        :return: an object of type input_entities[0] containing the merged vertices.
+        """
+        cls.validate_objects(input_entities)
+        kwargs.pop("workspace", None)
+
+        # create the vertices
+        vertices = np.vstack([input_entity.vertices for input_entity in input_entities])
+
+        # create an object of type input_entities[0]
+        output_type = type(input_entities[0])
+        output = output_type.create(workspace, vertices=vertices, **kwargs)
```

Comment:
This part should be implemented by the sub-class, as it will require more mechanics for objects will cells down the road.

Reply:
I changed merge_object to Points and let it as an abstract method in Base.

Reply:
I create a new function "create object": an abstract method.